WooCommerce/Classes/ViewRelated/Orders/Order Creation/ProductsSection/AddProductToOrderViewModel.swift

```diff
@@ -208,6 +254,30 @@ private extension AddProductToOrderViewModel {
             }
             .store(in: &subscriptions)
     }
+
+    /// Updates the product results predicate & triggers a new sync when search term changes
+    ///
+    func configureProductSearch() {
+        $searchTerm
+            .dropFirst() // Drop initial value
+            .debounce(for: .milliseconds(500), scheduler: DispatchQueue.main)
+            .removeDuplicates()
+            .sink { [weak self] newSearchTerm in
+                guard let self = self else { return }
+
+                if newSearchTerm.isNotEmpty {
+                    // When the search query changes, also includes the original results predicate in addition to the search keyword.
+                    let searchResultsPredicate = NSPredicate(format: "ANY searchResults.keyword = %@", newSearchTerm)
+                    let subpredicates = [self.resultsPredicate].compactMap { $0 } + [searchResultsPredicate]
```

Comment:
could this be
```swift
[self.resultsPredicate + searchResultsPredicate].compactMap { $0 } 
```
?

Reply:
Ah, they can't be combined with `+` but it could be this which is definitely easier to read:

```
[self.resultsPredicate, searchResultsPredicate].compactMap { $0 }
```

I'll make that change.